lib/compiler/test/beam_types_SUITE.erl

```diff
@@ -101,6 +101,16 @@ binary_absorption(Config) when is_list(Config) ->
     A = beam_types:meet(A, beam_types:join(A, B)),
     A = beam_types:join(A, beam_types:meet(A, B)),
 
+    %% Tests that the appendable flag behaves as intended.
```

Comment:
Feel free to update `beam_types_prop:gen_bs_matchable/0` too, the following ought to do it:

```erlang
gen_bs_matchable() ->
    oneof([?LET(Unit, range(1, 16), #t_bs_matchable{tail_unit=Unit}),
           ?LET(Unit, range(1, 16), #t_bs_context{tail_unit=Unit}),
           ?LET({Unit, Appendable}, {range(1, 16), boolean()},
                #t_bitstring{size_unit=Unit})]).
```